src/Chapter3.hs

```diff
@@ -560,22 +644,34 @@ introducing extra newtypes.
 🕯 HINT: if you complete this task properly, you don't need to change the
     implementation of the "hitPlayer" function at all!
 -}
+
+newtype Health    = MkHealth    { unHealth    :: Int } deriving (Show)
+newtype Armor     = MkArmor     { unArmor     :: Int } deriving (Show)
+newtype Attack    = MkAttack    { unAttack    :: Int } deriving (Show)
+newtype Dexterity = MkDexterity { unDexterity :: Int } deriving (Show)
+newtype Strength  = MkStrength  { unStrength  :: Int } deriving (Show)
+newtype Damage    = MkDamage    { unDamage    :: Int } deriving (Show)
+newtype Defense   = MkDefense   { unDefense   :: Int } deriving (Show)
+
 data Player = Player
-    { playerHealth    :: Int
-    , playerArmor     :: Int
-    , playerAttack    :: Int
-    , playerDexterity :: Int
-    , playerStrength  :: Int
+    { playerHealth    :: Health
+    , playerArmor     :: Armor
+    , playerAttack    :: Attack
+    , playerDexterity :: Dexterity
+    , playerStrength  :: Strength
     }
+    deriving (Show)
 
-calculatePlayerDamage :: Int -> Int -> Int
-calculatePlayerDamage attack strength = attack + strength
+calculatePlayerDamage :: Attack -> Strength -> Damage
+calculatePlayerDamage atk str = MkDamage (unAttack atk + unStrength str)
 
-calculatePlayerDefense :: Int -> Int -> Int
-calculatePlayerDefense armor dexterity = armor * dexterity
+calculatePlayerDefense :: Armor -> Dexterity -> Defense
+calculatePlayerDefense ar dex = MkDefense (unArmor ar * unDexterity dex)
 
-calculatePlayerHit :: Int -> Int -> Int -> Int
-calculatePlayerHit damage defense health = health + defense - damage
+calculatePlayerHit :: Damage -> Defense -> Health -> Health
+calculatePlayerHit dmg def h = MkHealth (unHealth h - inflictedDamagePoints)
+    where inflictedDamagePoints = max (unDamage dmg - unDefense def) 0  
```

Comment:
Nice work on newtypes! :medal_sports: Newtypes are one very important feature in Haskell :+1:
You can read more about the newtype pattern in our blog post: https://kowainik.github.io/posts/haskell-mini-patterns#newtype